### PR TITLE
A major re-organisation of Streambed

### PR DESCRIPTION
**Pull request by huntc**
I have re-organised Streambed so that it should become easier to navigate and maintain going forward. I have also leveraged Cargo's most recent capabilities to declare dependency versions in one place; the workspace configuration. Along the way I have updated its dependency chain.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,33 +1,37 @@
...
+    "streambed-test",
+    "streambed-vault",
+]
 
-[dev-dependencies]
-env_logger = "0.9"
-http = "0.2"
-hyper = { version = "0.14", default-features = false, features = ["tcp", "stream", "http1", "http2", "client", "server", "runtime"] }
-test-log = "0.2"
+[workspace.dependencies]
```
Comment (hseeberger) [inline Cargo.toml:12]: Good move!

File: README.md [hunk 2]
```diff
@@ -1,37 +1,64 @@
...
+Streambed is a curated set of dependencies and a toolkit for writing asynchronous event-driven services that aim to run on the 
+smallest "std" targets supported by Rust. Streambed components use a single core MIPS32 OpenWrt device running at around 500MHz 
+and 128MiB as a baseline target.
 
-Efficient
----
+A commit log modelled on Apache Kafka is provided, along with partial and extended support for the Kafka HTTP API.
 
-Streambed based applications are designed to run at the edge on embedded computers as well as in the cloud and so efficient CPU and memory usage are of primary concern.
+A secret stored modelled on Hashicorp Vault is provided along with partial support for the Vault HTTP API.
```
Comment (stevewillcock) [inline README.md:11]: "store", not "stored"?

File: README.md [hunk 3]
```diff
@@ -1,37 +1,64 @@
...
+Production services using the commit log and secret store have been shown to use less than 3MiB of resident memory
+while also offering good performance.
 
-Security is another primary consideration throughout the design of Streambed. For example, in the world of the Internet of Things, if an individual sensor becomes compromised then its effects can be minimized.
+## Streambed's traits
+### Event-driven services
 
-Built for integration
----
+Event-driven services promotes responsiveness as events can be pushed to where they need to be consumed; by the user
```
Comment (stevewillcock) [inline README.md:19]: "promote", not "promotes"

File: README.md [hunk 4]
```diff
@@ -1,37 +1,64 @@
...
----
+### Open source and open standards
 
 Streambed is entirely open source providing cost benefits, fast-time-to-market, the avoidance of vendor lock-in, improved security and more.
 
-Rust
----
+### Rust
 
 Streambed builds on Rust's traits of writing fast and efficient software correctly.
```
Comment (stevewillcock) [inline README.md:45]: "traits" may be a confusing word here as that means something specific in the Rust language - perhaps something more Rust-agnostic like "Streambed builds on the strengths of the Rust language, enabling us to write fast and efficient software correctly." Or maybe the use of the word "trait" was on-purpose :)

**Review by stevewillcock — APPROVED** [review]
Looks good!

File: README.md (hunk 2 above)
Comment (huntc) [reply README.md:11]: Thanks

File: README.md (hunk 4 above)
Comment (huntc) [reply README.md]: ok, eliminated the use of the word trait.

File: Cargo.toml (hunk 1 above)
Comment (huntc) [reply Cargo.toml:12]: Thanks!

**Pull request merged**